### PR TITLE
PAT-484 report Pathfinder queue status in health checks

### PR DESCRIPTION
**Pull request by timharrison-moj**
This PR grew some legs, so as well as basic queue health checks it also updates the README.md file with detailed instructions on the variety of ways to setup and run Pathfinder and its dependencies locally. As part of this I have rationalised the docker-compose files into -minimal, -full, -test and -pathfinder flavours to try an match the way we work. 

I identified some follow-on work required, so will raise these as different tickets:

* Replace minio with localstack for AWS S3 mocking
* Add the DLQ secrets to K8s secrets in Terraform, helm values and config.js so we can use them to query the number of messages on the DLQs for offender and tact event queues.
* Automatically create queues & DLQ queues in docker-compose to save on manual configuration. The instructions for creating these are in the README.md, but better to automate what we can.

### CODE REVIEW

File: docker-compose-full.yml [hunk 1]
```diff
@@ -83,40 +83,33 @@ services:
...
-    - ELITE2API_ENDPOINT_URL=http://elite2-api:8080
-    - GLOBAL_SEARCH_URL=http://localhost:3002/global-search
-    - DPS_QUICK_LOOK_URL=http://localhost:3001/offenders
-    - NODE_ENV=development
+      - SERVICES=${SERVICES-sqs}
+      - DEBUG=${DEBUG- }
+      - DATA_DIR=${DATA_DIR- }
+      - PORT_WEB_UI=${PORT_WEB_UI-8088}
+      - LAMBDA_EXECUTOR=${LAMBDA_EXECUTOR- }
+      - KINESIS_ERROR_PROBABILITY=${KINESIS_ERROR_PROBABILITY- }
```
Comment (garethmdavies) [inline docker-compose-full.yml]: Worth trying if these two lines can be removed.

File: docker-compose-full.yml [hunk 2]
```diff
@@ -83,40 +83,33 @@ services:
...
+      - DEFAULT_REGION=eu-west-2
+      - DOCKER_HOST=localhost
+    volumes:
+      - "${TMPDIR:-/tmp/localstack}:/tmp/localstack"
```
Comment (garethmdavies) [inline docker-compose-full.yml]: This doesn't work on OSX: Alternative to try is: - './.localstack:/tmp/localstack'

File: docker-compose-minimal.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      - SERVICES=${SERVICES-sqs}
+      - DEBUG=${DEBUG- }
+      - DATA_DIR=${DATA_DIR- }
+      - PORT_WEB_UI=${PORT_WEB_UI-8088}
+      - LAMBDA_EXECUTOR=${LAMBDA_EXECUTOR- }
+      - KINESIS_ERROR_PROBABILITY=${KINESIS_ERROR_PROBABILITY- }
+      - DEFAULT_REGION=eu-west-2
+      - DOCKER_HOST=localhost
+    volumes:
+      - "${TMPDIR:-/tmp/localstack}:/tmp/localstack"
```
Comment (garethmdavies) [inline docker-compose-minimal.yml]: Same issue here with OSX: try - './.localstack:/tmp/localstack'

File: server/data/healthcheck.js [hunk 4]
```diff
@@ -45,7 +45,46 @@ function serviceCheckFactory(name, url) {
...
+
+          if (isNilOrEmpty(attributes.RedrivePolicy)) {
+            reject({ status: 'ERROR', err: 'No DLQ is configured' })
+          }
+
+          const messagesOnQueue = attributes.ApproximateNumberOfMessages || 0
+          const messagesInFlight = attributes.ApproximateNumberOfMessagesNotVisible || 0
+          const messagesDelayed = attributes.ApproximateNumberOfMessagesDelayed || 0
+
+          // TODO: Add check for number of messages on the DLQ related to this queue - needs terraform to expose secrets in K8s
```
Comment (lbennett-moj) [inline server/data/healthcheck.js]: we do have the dlq secrets unless I'm misunderstanding

Comment (timharrison-moj) [reply server/data/healthcheck.js]: As discussed outside of here, you're right - they are pushed into k8s as secrets so its just that the app is not configuring these in its helm_deploy/pathinder/_envs.tpl nor getting these into server/config.js

File: docker-compose-full.yml (hunk 2 above)
Comment (timharrison-moj) [reply docker-compose-full.yml]: Altered this, and tested ok on Unbuntu.

File: docker-compose-minimal.yml (hunk 3 above)
Comment (timharrison-moj) [reply docker-compose-minimal.yml]: Altered this, and tested ok on Unbuntu. Now committed on branch.

**Pull request merged**